apps/mail/tasks.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import json
+import requests
+
+from django.conf import settings
+
+from workers import task
+
+
+log = logging.getLogger(__name__)
+
+
+@task()
+def send_email(mail_id):
+    """
+    Do not call this task directly. Instead use a subclass of MailBase.
+    """
+    from apps.mail.models import Mail
+
+    mail = Mail.objects.get(pk=mail_id)
+
+    if not settings.SEND_MAIL:
+        log.info('SEND_MAIL is not True! {}'.format(mail))
+        mail.status = Mail.TEST
+        mail.save()
+        return
+
+    headers = {
+        'authorization': 'Bearer {0}'.format(settings.SENDGRID_API_KEY),
+        'content-type': 'application/json'
+    }
+
+    to_email = getattr(mail.user, mail.user.get_email_field_name())
+    try:
+        to_name = u.get_full_name()
```

Comment:
I think this should be `mail.user.get_full_name()`?

Reply:
It's in a `try:` block so it's okay